### PR TITLE
Fix incorrect non-local diagnostic classification for VB symbol analyzers

### LINKED ISSUE #68654: Incorrect non-local diagnostic classification for VB symbol analyzer
Below test shows an analyzer that registers symbol action for `SymbolKind.NamedType` and reports diagnostics on field symbols within the named type.

**Expected:**
Reported diagnostic is classified as a local semantic diagnostic, which is displayed in live analysis and supports code fixes

**Actual:**
Reported diagnostic is classified as a non-local compilation diagnostic outside the scope of the callback symbol, which is not displayed in live analysis and does not support getting fixed by code fixers

```vb
        <Fact>
        Public Async Function TestAnalyzerLocalDiagnosticsWhenReportedOnFieldSymbol() As Task
            Dim source = "
Public Class Outer
    Public Enum E1
        A1 = 0
    End Enum
End Class

Public Enum E2
    A2 = 0
End Enum"

            Dim compilation = CreateCompilation(source)
            compilation.VerifyDiagnostics()

            Dim tree = compilation.SyntaxTrees(0)
            Dim analyzer = New NamedTypeFieldSymbolAnalyzer()
            Dim compilationWithAnalyzers = compilation.WithAnalyzers(ImmutableArray.Create(Of DiagnosticAnalyzer)(analyzer), AnalyzerOptions.Empty)
            Dim result = Await compilationWithAnalyzers.GetAnalysisResultAsync(CancellationToken.None)

            Dim localSemanticDiagnostics = result.SemanticDiagnostics(tree)(analyzer)
            localSemanticDiagnostics.Verify(
                Diagnostic("ID0001", "A1 = 0").WithLocation(4, 9),
                Diagnostic("ID0001", "A2 = 0").WithLocation(9, 5))

            Assert.Empty(result.CompilationDiagnostics)
        End Function

        <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
        Private Class NamedTypeFieldSymbolAnalyzer
            Inherits DiagnosticAnalyzer

            Public Shared ReadOnly Descriptor As New DiagnosticDescriptor("ID0001", "Title", "Message", "Category", defaultSeverity:=DiagnosticSeverity.Warning, isEnabledByDefault:=True)

            Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor)
                Get
                    Return ImmutableArray.Create(Descriptor)
                End Get
            End Property

            Public Overrides Sub Initialize(context As AnalysisContext)
                context.RegisterSymbolAction(Sub(symbolContext As SymbolAnalysisContext)
                                                 Dim namedType = DirectCast(symbolContext.Symbol, INamedTypeSymbol)
                                                 For Each field In namedType.GetMembers().OfType(Of IFieldSymbol)
                                                     If Not field.IsImplicitlyDeclared Then
                                                         Dim diag = CodeAnalysis.Diagnostic.Create(Descriptor, field.DeclaringSyntaxReferences(0).GetLocation())
                                                         symbolContext.ReportDiagnostic(diag)
                                                     End If
                                                 Next
                                             End Sub,
                    SymbolKind.NamedType)
            End Sub
        End Class
```

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #68654

Fix is to update GetTopmostNodeForDiagnosticAnalysis VB override to handle EnumStatementSyntax to EnumBlockSyntax mapping similar to the TypeStatementSyntax to TypeBlockSyntax mapping that was already being done.

Prior to the fix, the diagnostic is classified as a non-local compilation diagnostic and fails the added unit test.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb [hunk 1]
```diff
@@ -3533,6 +3533,10 @@ _Default:
                         If declaringSyntax.Parent IsNot Nothing AndAlso TypeOf declaringSyntax.Parent Is TypeBlockSyntax Then
                             Return declaringSyntax.Parent
                         End If
+                    ElseIf TypeOf declaringSyntax Is EnumStatementSyntax Then
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb:3536]: >ElseIf TypeOf declaringSyntax Is EnumStatementSyntax Then [](http://example.com/codeflow?start=20&length=57)

Was the issue specific to Enum types? The issue doesn't mention that.

File: src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb [hunk 2]
```diff
@@ -674,5 +674,60 @@ End Class"
                 reportDiagnostic(CodeAnalysis.Diagnostic.Create(Descriptor, location, containingSymbol.Name))
             End Sub
         End Class
+
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/68654")>
+        Public Async Function TestAnalyzerLocalDiagnosticsWhenReportedOnFieldSymbol() As Task
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb:679]: >TestAnalyzerLocalDiagnosticsWhenReportedOnFieldSymbol [](http://example.com/codeflow?start=30&length=53)

Is it possible to add similar test for C#?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb (hunk 1 above)
Comment (mavasani) [reply src/Compilers/VisualBasic/Portable/Compilation/SemanticModel.vb:3536]: Yes, I identified that after further investigation.

File: src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb (hunk 2 above)
Comment (mavasani) [reply src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb:679]: Yes, definitely. Will do.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Review by 333fred — APPROVED** [review]

**Pull request merged**